src/daemonizer/posix_fork.cpp

```diff
@@ -21,15 +21,42 @@
 namespace posix {
 
 namespace {
-  void quit(std::string const & message)
+  void quit(const std::string & message)
   {
     LOG_ERROR(message);
     throw std::runtime_error(message);
   }
 }
 
-void fork()
+void fork(const std::string & pidfile)
 {
+  // If a PID file is specified, we open the file here, because
+  // we can't report errors after the fork operation.
+  // When we fork, we close thise file in each of the parent
+  // processes.
+  // Only in the final child process do we write the PID to the
+  // file (and close it).
+  std::ofstream pidofs;
+  if (! pidfile.empty ())
+  {
+	int oldpid;
+    std::ifstream pidrifs;
+    pidrifs.open(pidfile, std::fstream::in);
+    if (! pidrifs.fail())
+    {
+	  if (pidrifs >> oldpid && oldpid > 1 && kill(oldpid, 0) == 0)
+      {
+        quit("PID file " + pidfile + " already exists and the PID therein is valid");
```

Comment:
Yes but a user who is starting and stopping domains often and often elects to use pid files, may end up with situations where they forget to change the pid file name. Then by your code the new daemon will not launch, even though they may not care to much about the pid file. Listen, I am not saying this is a major problem, rather it would be a preferable and trivial change to just remove the quit and replace with a warning message. The only time I have seen daemon code that exits at startup is when the reason it exits is because there is no way it can possibly run, which is not the case with this pid use-case.